test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -42,52 +42,85 @@ timestamp_manager::get_timestamp_manager()
-                  << " must not be later than stable timestamp " << new_stable_ts << "."
-                  << std::endl;
-        return 1;
+    /* If config has stable timestamp */
+    if (has_stable) {
+        /* Cannot proceed any further, if stable timestamp value <= 0 */
+        if ((int64_t)new_stable_ts <= 0)
+            return (false);
+        /* It is a no-op to set the new stable timestamps behind the current stable timestamp. */
+        if (new_stable_ts <= conn->get_stable_ts())
+            has_stable = false;
     }
 
-    return 0;
-}
+    /* No need to validate timestamps if stable or/and oldest were behind the global values. */
+    if (!has_oldest && !has_stable)
+        return (true);
 
-int
-timestamp_manager::validate_stable_ts(uint64_t new_stable_ts, uint64_t new_oldest_ts)
-{
-    connection_simulator *conn = &connection_simulator::get_connection();
+    /* No need to validate timestamps if there is no new and no current oldest timestamp. */
+    if (!has_oldest && conn->get_oldest_ts() == 0)
+        return (true);
 
-    /* Stable timestamp can't move backward. */
-    if (new_stable_ts <= conn->get_stable_ts()) {
-        std::cout << "Stable timestamp cannot move backwards." << std::endl;
-        return 1;
-    }
+    /* No need to validate timestamps if there is no new and no current stable timestamp. */
+    if (!has_stable && conn->get_stable_ts() == 0)
+        return (true);
 
-    /* The oldest and stable timestamps must always satisfy the condition that oldest <= stable. */
-    if (new_oldest_ts > new_stable_ts) {
-        std::cout << "set_timestamp: oldest timestamp " << conn->get_oldest_ts()
-                  << " must not be later than stable timestamp " << new_stable_ts << "."
-                  << std::endl;
-        return 1;
-    }
+    /*
+     * If the oldest timestamp was not passed in the config or was behind the current oldest
+     * timestamp, modify the new_oldest_ts to the current oldest timestamp.
+     */
+    if ((!has_oldest && conn->get_oldest_ts() != 0))
+        new_oldest_ts = conn->get_oldest_ts();
+
+    /*
+     * If the stable timestamp was not passed in the config or was behind the current stable
+     * timestamp, modify the new_stable_ts to the current stable timestamp
+     */
+    if ((!has_stable && conn->get_stable_ts() != 0))
+        new_stable_ts = conn->get_stable_ts();
 
-    return 0;
+    /* Validate if stable is greater than the oldest timestamp. */
+    if (new_oldest_ts > new_stable_ts)
+        return (false);
+
+    return (true);
 }
 
-int
-timestamp_manager::validate_durable_ts(uint64_t new_durable_ts)
+bool
+timestamp_manager::validate_durable_ts(uint64_t &new_durable_ts, bool &has_durable)
 {
-    return 0;
+    /* If durable timestamp was not passed in the config, no validation is needed. */
+    if (!has_durable)
+        return (true);
+
+    /* Illegal timestamp value (if less than or equal to 0). */
+    if ((int64_t)new_durable_ts <= 0)
```

Comment:
This is not possible as it was passed as an `uint64_t`, it will never be negative. Furthermore, casting from `uint64_t` to `int64_t` does not seem correct without a better range check.

Reply:
This is intentional. 
If `-ve` value is converted to an unsigned int, it becomes a big value. 
It is possible to convert an unsigned int back to a signed int.

This is exactly what is being done here.